boot/freeldr/freeldr/arch/uefi/uefimem.c

```diff
@@ -303,10 +314,21 @@ UefiExitBootServices(VOID)
     }
 }
 
+
+BOOLEAN
+MempSetupPaging(IN PFN_NUMBER StartPage,
+                IN PFN_NUMBER NumberOfPages,
+                IN BOOLEAN KernelMapping);
+
+ULONG* StackPtrNew;
+ULONG* StackPtrEnd;
 VOID
 UefiPrepareForReactOS(VOID)
 {
+    ULONG StackVar = 0;
+    StackPtrNew = &StackVar;
+    StackPtrEnd = (&StackVar + 0x2000); /* Technically UEFI defines that stack space as 32kb*/
     UefiExitBootServices();
-    ExitStack = MmAllocateMemoryWithType(EXIT_STACK_SIZE, LoaderOsloaderStack);
-    EndofExitStack = (PVOID)((ULONG_PTR)ExitStack + EXIT_STACK_SIZE);
+    //TODO: Let's take another look at this
+    MempSetupPaging((ULONG_PTR)StackPtrEnd >> MM_PAGE_SHIFT, 0x2000 >> MM_PAGE_SHIFT, FALSE);
```

Comment:
This call cannot be done before the UefiExitBootServices call?

Reply:
Doesn’t matter either way

Reply:
I was asking, because maybe in one case you could move those StackPtr and StackPtrEnd variables into the function itself (instead of keeping them as globals)